src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.ProjectFormat/SdkStyleConversionSolutionWideStep.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.ProjectFormat
+{
+    public class SdkStyleConversionSolutionWideStep : UpgradeStep
+    {
+        private readonly TryConvertRunner _runner;
+
+        private IEnumerable<UpgradeStep>? _substeps;
+
+        public override string Description => $"Use the try-convert tool ({_runner.Path}{_runner.VersionString}) to convert the projects in a solution";
+
+        public override string Title => $"Convert projects in a solution to SDK style";
+
+        public override string Id => WellKnownStepIds.TryConvertProjectConverterStepId;
+
+        public override IEnumerable<string> DependsOn { get; } = new[]
+        {
+            WellKnownStepIds.EntrypointSelectionStepId,
+        };
+
+        public override IEnumerable<string> DependencyOf { get; } = new[]
+        {
+            WellKnownStepIds.CurrentProjectSelectionStepId,
+        };
+
+        public SdkStyleConversionSolutionWideStep(
+            TryConvertRunner runner,
+            ILogger<SdkStyleConversionSolutionWideStep> logger)
+            : base(logger)
+        {
+            _runner = runner ?? throw new ArgumentNullException(nameof(runner));
+        }
+
+        protected override Task<bool> IsApplicableImplAsync(IUpgradeContext context, CancellationToken token) => Task.FromResult(true);
```

Comment:
This is probably ok, but our other solution-wide steps (project selection, etc.) only return true if `context.CurrentProject` is null so that the step doesn't show up once the user is working on a specific project.

Reply:
ok. I've updated to follow that pattern